### PR TITLE
Chunked encoding for tasks APIs

### PR DESCRIPTION
**Pull request by DaveCTurner**
This response can reach many MiB in size in a large and busy cluster, let's use chunking here.

Relates #89838

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/collect/StreamUtils.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public class StreamUtils {
+
+    private StreamUtils() {
+        // no instances
+    }
+
+    @SafeVarargs
+    @SuppressWarnings("varargs")
+    public static <T> Stream<T> concat(Stream<T>... streams) {
+        if (streams.length == 0) {
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/common/collect/StreamUtils.java:22]: Why not just `return Arrays.stream(streams).flatMap(Function.identity());` ? We might even get into stack depth troubles if we use the concat like we do here?
The docs for concat state:

```
Use caution when constructing streams from repeated concatenation. Accessing an element of a deeply concatenated stream can result in deep call chains, or even StackOverflowError.
Subsequent changes to the sequential/parallel execution mode of the returned stream are not guaranteed to be propagated to the input streams.
```

**Review by original-brownbear — COMMENTED** [review]
Looks good, just one spot to fix I think :)

File: server/src/main/java/org/elasticsearch/common/collect/StreamUtils.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/common/collect/StreamUtils.java]: > We might even get into stack depth troubles

Touché :grin:

Yeah you're right `.flatMap(Function.identity())` is better here, no need for a utility to do this.

**Review by original-brownbear — APPROVED** [review]
LGTM :)

**Pull request merged**